### PR TITLE
Feat: Implementing test creation on test overlay

### PR DESCRIPTION
**Pull request by stephanwampouille**
This PR includes the new create tests button on the homepage for build. Create tests requires an input for the test name (rule) and then makes the call to create the test. 

Design will be revisited once Val has had a chance to create new ones

Screen Shot 2023-01-03 at 2 13 07 PM

### CODE REVIEW

**Comment by khyberspache** [comment]
Where do we want the create Test button to go once a test code is opened?

**Comment by stephanwampouille** [comment]
I am doing it now, but I think it will go on there side panel still, per Val's design

**Comment by stephanwampouille** [comment]
Per my latest commit I matched the func. with Val's design. I also added a state manager on the Panel itself to disable the test creation while it is fetching for tests. One thing to note is that the adjustable bar cuts into the panel, I can think of a few ways to remedy this visually but that is not included in this PR.

https://user-images.githubusercontent.com/84744117/210489175-305a69dc-42ae-4b71-bd5b-8a1ab6eecbd2.mov

**Comment by stephanwampouille** [comment]
@kenrick ready for review

**Review by khyberspache — APPROVED** [review]
Love it

File: apps/build/src/components/browser/create-test.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    try {
+      setIsLoading(true);
+      await createTest(
+        testId,
+        rule,
+        code,
+        serviceConfig,
+        new AbortController().signal
+      );
+      notifySuccess("Successfully created test");
```
Comment (kenrick) [inline apps/build/src/components/browser/create-test.tsx:45]: After create invalidate the tests query

```
const queryClient = useQueryClient()

queryClient.invalidateQueries({ queryKey: ['tests', serviceConfig] })
```

This way is shows in the tests list.

File: apps/build/src/components/browser/create-test.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  };
+
+  return (
+    <>
+      <div className={styles.title}>
+        <span>Security Tests</span>
+        <IconButton
+          onClick={() => setCreateVisible(!createVisible)}
+          className={styles.create}
+          icon={<PlusIcon />}
```
Comment (kenrick) [inline apps/build/src/components/browser/create-test.tsx:60]: Show an X when then create is showing so the user knows they can close it

```suggestion
          icon={createVisible ? <CloseIcon> :  <PlusIcon />}
```

File: apps/build/src/components/browser/create-test.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import { useEffect, useState } from "react";
+import shallow from "zustand/shallow";
+import { getLanguage } from "../../lib/lang";
+import * as uuid from "uuid";
+import styles from "./create-test.module.css";
+
+const CreateTest: React.FC<{ testsLoading: boolean }> = ({ testsLoading }) => {
+  const serviceConfig = useAuthStore(select("host", "credentials"), shallow);
+  const [rule, setRule] = useState("");
+  const [isLoading, setIsLoading] = useState(testsLoading);
```
Comment (kenrick) [inline apps/build/src/components/browser/create-test.tsx:20]:
```suggestion
  const [isLoading, setIsLoading] = useState(false);
```

File: apps/build/src/components/browser/create-test.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  const [createVisible, setCreateVisible] = useState(false);
+
+  useEffect(() => {
+    if (testsLoading === false) {
+      setIsLoading(false);
+    }
+  }, [testsLoading]);
```
Comment (kenrick) [inline apps/build/src/components/browser/create-test.tsx:27]:
```suggestion
```

File: apps/build/src/components/browser/create-test.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+  return (
+    <>
+      <div className={styles.title}>
+        <span>Security Tests</span>
+        <IconButton
+          onClick={() => setCreateVisible(!createVisible)}
+          className={styles.create}
+          icon={<PlusIcon />}
+          loading={isLoading}
```
Comment (kenrick) [inline apps/build/src/components/browser/create-test.tsx:61]:
```suggestion
          loading={testsLoading || isLoading}
```

File: apps/build/src/components/browser/create-test.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+          disabled={isLoading}
```
Comment (kenrick) [inline apps/build/src/components/browser/create-test.tsx:62]:
```suggestion
          disabled={testsLoading || isLoading}
```

File: apps/build/src/components/browser/verified-tests.tsx [hunk 7]
```diff
@@ -22,19 +22,27 @@ import {
...
 const VerifiedTests: React.FC = () => {
   const { data, isLoading } = useTests();
   const testIds = useMemo(() => new Set(data?.map((t) => t.id)), [data]);
+  const [testsLoading, setTestsLoading] = useState(true);
+
+  useEffect(() => {
+    if (isLoading === false) {
+      setTestsLoading(false);
+    }
+  }, [isLoading]);
```
Comment (kenrick) [inline apps/build/src/components/browser/verified-tests.tsx:42]: This is not being used
```suggestion
```

**Review by kenrick — CHANGES_REQUESTED** [review]
Few changes @stephanwampouille

**Comment by kenrick** [comment]
@stephanwampouille Also when the test is created we should open it in the editor

**Comment by stephanwampouille** [comment]
@kenrick ready for review

**Review by kenrick — APPROVED** [review]
Looks good!

**Pull request merged**